src/features/Entities/Setting/SettingSlice.ts

```diff
@@ -1,32 +1,213 @@
+  payload: UpdateSyncAuthUsernameActionPayload;
+}
+
+export interface UpdateSyncAuthPasswordActionPayload {
+  password: string;
+}
+
+export interface UpdateSyncAuthPasswordAction {
+  type: string;
+  payload: UpdateSyncAuthPasswordActionPayload;
+}
+
+export interface UpdateSyncLastSyncActionPayload {
+  lastSync: number;
+}
+
+export interface UpdateSyncLastSyncAction {
+  type: string;
+  payload: UpdateSyncLastSyncActionPayload;
+}
+
+export interface UpdateSyncIsPassiveSyncEnabledActionPayload {
+  isPassiveSyncEnabled: boolean;
+}
+
+export interface UpdateSyncIsPassiveSyncEnabledAction {
+  type: string;
+  payload: UpdateSyncIsPassiveSyncEnabledActionPayload;
+}
+
+export interface PrepareActionReturn<SomePayload> {
+  payload: SomePayload;
 }
 
 const initialState: SettingSliceState = {
-  [SETTING.INT.DEFAULT_LOG_INTERVAL]: null,
+  [SettingSlice.Bluetooth]: {
+    [BluetoothSettingSlice.DefaultLogInterval]: null
+  },
+  [SettingSlice.Sync]: {
+    [SyncSettingSlice.AuthUrl]: null,
+    [SyncSettingSlice.AuthUsername]: null,
+    [SyncSettingSlice.AuthPassword]: null,
+    [SyncSettingSlice.SensorUrl]: null,
+    [SyncSettingSlice.TemperatureLogUrl]: null,
+    [SyncSettingSlice.TemperatureBreachUrl]: null,
+    [SyncSettingSlice.LastSync]: null,
+    [SyncSettingSlice.IsPassiveSyncEnabled]: null
+  }
 };
 
+const parseSettingSlice = (setting: SettingEntity) => {
+  switch (setting.key) {
+    case SETTING.BLUETOOTH.DEFAULT_LOG_INTERVAL:
+      return SettingSlice.Bluetooth;
+    case SETTING.SYNC.AUTH_URL:
+    case SETTING.SYNC.AUTH_USERNAME:
+    case SETTING.SYNC.AUTH_PASSWORD:
+    case SETTING.SYNC.SENSOR_URL:
+    case SETTING.SYNC.TEMPERATURE_LOG_URL:
+    case SETTING.SYNC.TEMPERATURE_BREACH_URL:
+    case SETTING.SYNC.LAST_SYNC:
+    case SETTING.SYNC.IS_PASSIVE_SYNC_ENABLED:
+      return SettingSlice.Sync;
+    default:
+      return SettingSlice.Sync;
+  }
+}
+
+const parseSettingKey = (setting: SettingEntity) => {
+  switch (setting.key) {
+    case SETTING.BLUETOOTH.DEFAULT_LOG_INTERVAL:
+      return BluetoothSettingSlice.DefaultLogInterval;
+    case SETTING.SYNC.AUTH_URL:
+      return SyncSettingSlice.AuthUrl;
+    case SETTING.SYNC.AUTH_USERNAME:
+      return SyncSettingSlice.AuthUsername;
+    case SETTING.SYNC.AUTH_PASSWORD:
+      return SyncSettingSlice.AuthPassword;
+    case SETTING.SYNC.SENSOR_URL:
+      return SyncSettingSlice.SensorUrl;
+    case SETTING.SYNC.TEMPERATURE_LOG_URL:
+      return SyncSettingSlice.TemperatureLogUrl;
+    case SETTING.SYNC.TEMPERATURE_BREACH_URL:
+      return SyncSettingSlice.TemperatureBreachUrl;
+    case SETTING.SYNC.LAST_SYNC:
+      return SyncSettingSlice.LastSync;
+    case SETTING.SYNC.IS_PASSIVE_SYNC_ENABLED:
+      return SyncSettingSlice.IsPassiveSyncEnabled;
+    default:
+      return setting.key;
+  }
+}
+
+const parseValue = (setting: SettingEntity) => {
+  try {
+    return JSON.parse(setting.value);
+  } catch {
+    return setting.value;
+  }
+}
```

Comment:
Definitely a better way to do this, but PRing in interests of not spending too long on non-functional changes. Thinking the idiomatic way is probably to combine reducers for the different setting slices?